src/main.h

```diff
@@ -52,6 +52,17 @@ CommandLineOptions;
 extern CommandLineOptions cl_options;
 
 
+/* Information about command line entries that can not be stored in GOptionEntry. */
+typedef struct
+{
+	gboolean persistent;  /* The option should be passed to "New Window" */
+}
+GeanyOptionEntryAux;
+
+extern GOptionEntry optentries[];
+extern GeanyOptionEntryAux optentries_aux[];
```

Comment:
The argv parsing happens in GLib. A function may be executed at any time after that. We can't simply get them from the original argv.

I'd rather export a function returning the reproduced GPtrArray (or simply char **). This way, we can extend it reproduce different sets of options, should need be. And GeanyOptionEntryAux is trivial to extend...